XMonad/Layout/IndependentScreens.hs

```diff
@@ -175,14 +242,40 @@ whenCurrentOn s pp = pp
 -- | If @vSort@ is a function that sorts 'WindowSpace's with virtual names, then @marshallSort s vSort@ is a function which sorts 'WindowSpace's with physical names in an analogous way -- but keeps only the spaces on screen @s@.
 marshallSort :: ScreenId -> ([WindowSpace] -> [WindowSpace]) -> ([WindowSpace] -> [WindowSpace])
 marshallSort s vSort = pScreens . vSort . vScreens where
-    onScreen ws = unmarshallS (tag ws) == s
+    onScreen ws = unmarshallS (W.tag ws) == s
     vScreens    = map unmarshallWindowSpace . filter onScreen
     pScreens    = map (marshallWindowSpace s)
 
+-- | Like 'marshallSort', but don't convert 'WindowSpace'\'s physical
+-- names back to virtual names.
+marshallSort' :: ScreenId -> ([WindowSpace] -> [WindowSpace]) -> ([WindowSpace] -> [WindowSpace])
+marshallSort' s vSort = vSort . vScreens where
+   onScreen ws = unmarshallS (W.tag ws) == s
+   vScreens    = filter onScreen
+
 -- | Convert the tag of the 'WindowSpace' from a 'VirtualWorkspace' to a 'PhysicalWorkspace'.
 marshallWindowSpace   :: ScreenId -> WindowSpace -> WindowSpace
 -- | Convert the tag of the 'WindowSpace' from a 'PhysicalWorkspace' to a 'VirtualWorkspace'.
 unmarshallWindowSpace :: WindowSpace -> WindowSpace
 
-marshallWindowSpace s ws = ws { tag = marshall s  (tag ws) }
-unmarshallWindowSpace ws = ws { tag = unmarshallW (tag ws) }
+marshallWindowSpace s ws = ws { W.tag = marshall s  (W.tag ws) }
+unmarshallWindowSpace ws = ws { W.tag = unmarshallW (W.tag ws) }
+
+
+-- $logHook
```

Comment:
Rename this section to `"XMonad.Actions.CopyWindow" integration` and move the relevant new functions there, please.

Reply:
I have done so.